neuralmonkey/decoders/sequence_labeler.py

```diff
@@ -139,3 +140,59 @@ def feed_dict(self, dataset: Dataset, train: bool = False) -> FeedDict:
             fd[self.train_weights] = paddings.T
 
         return fd
+
+
+class CRFLabeler(SequenceLabeler):
+    """CRF Classifier assing a label to each encoder's state."""
+
+    def __init__(self,
+                 name: str,
+                 encoder: Union[RecurrentEncoder, SentenceEncoder],
+                 vocabulary: Vocabulary,
+                 data_id: str,
+                 dropout_keep_prob: float = 1.0,
+                 save_checkpoint: Optional[str] = None,
+                 load_checkpoint: Optional[str] = None) -> None:
+        SequenceLabeler.__init__(self, name, encoder, vocabulary, data_id,
+                                 dropout_keep_prob, save_checkpoint,
+                                 load_checkpoint)
+
+        self.transition_params = tf.get_variable(
+            "transition_params",
+            [len(self.vocabulary) - 3, len(self.vocabulary) - 3],
+            initializer=tf.glorot_uniform_initializer())
+
+    @tensor
+    def logprobs(self) -> tf.Tensor:
+        raise NotImplementedError("Logarithmic probabilities are not "
+                                  "supported by the CRFLabeler")
```

Comment:
Tohle je divný. Musí bejt CRF labeler potomkem sequence labeleru? Nemůže to bejt spíš komponenta nad tim labelerem, která bude používat jeho logity? V konstruktoru si ten labeler může vytvořit a mít ho jako atribut, ne?